src/Entity/ReactionRuleStorage.php

```diff
@@ -61,6 +69,19 @@ public function __construct(EntityTypeInterface $entity_type, ConfigFactoryInter
   }
 
   /**
+   * Gets the event manager.
+   *
+   * @return \Drupal\rules\Core\RulesEventManager
+   *   The event manager.
+   */
+  protected function eventManager() {
+    if (!$this->eventManager) {
+      $this->eventManager = new RulesEventManager($this->moduleHandler());
```

Comment:
nope - this must use either proper depdency injection or the pattern as e.g. \Drupal\Core\Form\FormBase::getRouteMatch. As the class already uses proper constructor injection, we should use that here also. In addition, this should be getEventManager().